R/getCPR.R

```diff
@@ -1,44 +1,42 @@
 #' Get cells per region
-#' 
-#' Counts how many cells each region has and returns it as vector
-#' 
-#' 
+#'
+#' Counts how often each element of the provided subdimension exists in the given
+#' data set. Originally created to count the number of cells in a region (this
+#' is also where its name originates from) it can now be used to count elements
+#' of any subdimension via the dim argument.
+#'
 #' @param x MAgPIE object or a resolution written as numeric (currently only
 #' data for 0.5 degree resolution is available).
+#' @param dim Dimension for which the items should be returned. Either number or
+#' name of dimension or a vector of these (in case of a vector all subimensions
+#' must belong to the same main dimension!). See \code{\link{dimCode}} for more details.
 #' @return cells per region
 #' @author Jan Philipp Dietrich
 #' @seealso \code{\link{getRegions}}, \code{\link{read.magpie}},
 #' \code{\link{write.magpie}}
 #' @examples
-#' 
-#' # a <- read.magpie("example.mz")
-#' # getCPR(a)
+#'
 #' getCPR(0.5)
-#' 
-#' @export getCPR
-getCPR <- function(x) {
-  if(!is.magpie(x)) {
-    if(x==0.5) {
-      region.code <- magclassdata$half_deg$region
-      cpr <- rep(0,length(levels(region.code)))
-      names(cpr) <- levels(region.code)
-      for(region in names(cpr)) {
-       cpr[region] <- length(grep(region,region.code))  
-      }
-    } else {
-      stop(paste("No cells-per-region information available for resolution",x))
-    }
-  } else {
-    # try to reorder by cell/cluster number
-    cells <- try(as.integer(getItems(x,1.2)),silent = TRUE)
-    if(class(cells)!="try-error" && !anyNA(cells)) x <- x[order(cells),,]
-    
-    region_names <- getRegions(x)
-    cpr <- rep(0,length(region_names))
-    names(cpr) <- region_names
-    for(region in region_names) {
-      cpr[region] <- length(grep(region,dimnames(x)[[1]]))  
-    }
-  }
-  return(cpr)
+#' a <- maxample("animal")
+#' getCPR(a, dim = "color")
+#' getCPR(a, dim = 3.2)
+#' getCPR(a, dim = "country")
+#' getCPR(a, dim = c("color", "species"))
+#' @export
+
+getCPR <- function(x, dim = 1.1) {
+ if (is.magpie(x)) {
```

Comment:
Indentation looks off, it's only one space, but should be 2 I think. Fix with `ctrl+i` + `lucode2::autoFormat`? It could be that RStudio automatically tries to infer indentation from the file, but I think you can force RStudio to always use 2 spaces for an indentation level